src/core/quantum_gate.jl

```diff
@@ -1,13 +1,13 @@
 """
-    AbstractGate
+    AbstractGateSymbol
 
-A `Gate` is an instantiation of an `AbstractGate`, which can be added to a `QuantumCircuit` in order to apply an operator to one or more `target` qubits.
-`AbstractGate` is useful to dispatch all `Gates` to default implementation of functions such as get_connected_qubits(). 
+A `Gate` is an instantiation of an `AbstractGateSymbol`, which can be added to a `QuantumCircuit` in order to apply an operator to one or more `target` qubits.
+`AbstractGateSymbol` is useful to dispatch all `Gates` to default implementation of functions such as get_connected_qubits(). 
 Those functions are then specialized for `Gates` requiring a different implementation. 
 
-`AbstractGate` is an abstract type, which means that it cannot be instantiated. 
-Instead, each concrete type of `Gate` is a struct which is a subtype of `AbstractGate`.
-Each descendant of `AbstractGate` must have at least the following fields:
+`AbstractGateSymbol` is an abstract type, which means that it cannot be instantiated. 
+Instead, each concrete type of `Gate` is a struct which is a subtype of `AbstractGateSymbol`.
+Each descendant of `AbstractGateSymbol` must have at least the following fields:
```

Comment:
I've updated the documentation since neither of those fields is required, and only the `get_operator` and `get_num_connected_qubits` functions are required.